src/services/TwoFactorService.ts

```diff
@@ -0,0 +1,116 @@
+import { authenticator } from 'otplib';
+import * as QRCode from 'qrcode';
+import { User } from '../entities/User';
+import { initORM } from '../db';
+
+class TwoFactorService {
+  private readonly appName = 'Ghost Drive';
+
+  /**
+   * Generate a new 2FA secret for a user
+   */
+  generateSecret(): string {
+    return authenticator.generateSecret();
+  }
+
+  /**
+   * Generate QR code data URL for 2FA setup
+   */
+  async generateQRCode(username: string, secret: string): Promise<string> {
+    const otpauth = authenticator.keyuri(username, this.appName, secret);
+    return await QRCode.toDataURL(otpauth);
+  }
+
+  /**
+   * Verify a 2FA token
+   */
+  verifyToken(token: string, secret: string): boolean {
+    try {
+      return authenticator.check(token, secret);
+    } catch (error) {
+      return false;
+    }
+  }
+
+  /**
+   * Setup 2FA for a user (generate secret and QR code)
+   */
+  async setup2FA(userId: number): Promise<{ secret: string; qrCode: string }> {
+    const db = await initORM();
+    const userInDb = await db.user.findOne({ id: userId });
+    if (!userInDb) {
+      throw new Error("User not found");
+    }
+
+    const secret = this.generateSecret();
+    const qrCode = await this.generateQRCode(userInDb.username, secret);
+    
+    // Store the secret temporarily (not enabled yet)
+    userInDb.twoFactorSecret = secret;
+    await db.em.persistAndFlush(userInDb);
+
+    return { secret, qrCode };
```

Comment:
WTF secret is saved in Backend only user(or FE) just need the qrcode data so user can use 2FA app to scan and setup 2FAFA